### DISCUSSION TITLE
useSWR data not fetching...

### DISCUSSION THREAD

**Discussion by AbhayaShankar** [vercel/swr#2590] [General] [answered]
import useSWR from "swr";  
const { data, error } = useSWR(
    "https://next-js-45b5e-default-rtdb.firebaseio.com/events.json"
  );

  console.log(data);
  
  why is data logging as undefined...
  I have correctly installed the package, importing it correctly
  What could be the reason data is not getting fetched?

**Comment by koba04** [comment, marked as answer]
You have to specify a fetcher function as the argument.
https://swr.vercel.app/docs/data-fetching

**Reply by AbhayaShankar** [reply]
Thanks kobayashi, Got it now.
